### PR TITLE
CAUSEWAY-3358: Allow SAFE semantics actions to be invoked with only VIEWING permission

### PR DESCRIPTION
**Pull request by andi-huber**
- [x] implement
- [x] add test cases

### CODE REVIEW

File: core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java [hunk 1]
```diff
@@ -115,6 +113,14 @@ public CausewayConfiguration(final ConfigurableEnvironment environment) {
     private final Security security = new Security();
     @Data
     public static class Security {
+
+        /**
+         * If set, allows <i>Actions</i> with SAFE Semantics to be invoked with only VIEWING permissions.
```
Comment (danhaywood) [inline core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java:118]: trivial : @link to `@Action` annotation and `Semantics` enum?

Comment (andi-huber) [reply core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java:118]: yup, good point

File: core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java [hunk 2]
```diff
@@ -115,6 +113,14 @@ public CausewayConfiguration(final ConfigurableEnvironment environment) {
...
+         * <p>
+         * default: false
+         */
+        private boolean allowActionsWithSafeSemanticsToBeInvokedWithOnlyViewingPermission = false;
```
Comment (danhaywood) [inline core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java:122]: perhaps a little bit shorter?  something like:

`actionsWithSafeSemanticsRequireOnlyViewingPermission`

Comment (andi-huber) [reply core/config/src/main/java/org/apache/causeway/core/config/CausewayConfiguration.java:122]: agreed

File: regressiontests/stable-interact/src/test/java/org/apache/causeway/testdomain/interact/ActionInteractionTest_usingAllowSafeSemantics.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import lombok.val;
+
+@SpringBootTest(
+        classes = {
+                Configuration_headless.class,
+                Configuration_usingInteractionDomain.class,
+                ActionInteractionTest_usingAllowSafeSemantics.AuthorizorDenyUse.class
+        },
+        properties = {
+                "causeway.security.allowActionsWithSafeSemanticsToBeInvokedWithOnlyViewingPermission=TRUE",
```
Comment (danhaywood) [inline regressiontests/stable-interact/src/test/java/org/apache/causeway/testdomain/interact/ActionInteractionTest_usingAllowSafeSemantics.java:53]: Do we need one further test to confirm that when not set, then the behaviour still requires CHANGING perm?

**Comment by danhaywood** [comment]
Thanks for this @andi-huber  ... added a few small comments against the diff.  Otherwise, really nice!

File: regressiontests/stable-interact/src/test/java/org/apache/causeway/testdomain/interact/ActionInteractionTest_usingAllowSafeSemantics.java (hunk 3 above)
Comment (andi-huber) [reply regressiontests/stable-interact/src/test/java/org/apache/causeway/testdomain/interact/ActionInteractionTest_usingAllowSafeSemantics.java:53]: good point, lets add that

**Pull request merged**